web/src/test/auto/integrated/test_utils.js

```diff
@@ -79,44 +74,83 @@ export function setupKMW(kmwOptions, timeout) {
  *
  * @param {*} src       The source script's (relative) path on the test server.
  * @param {*} timeout
- * @param {*} functor   A callback to handle the script element.
  * @returns
  */
-export function setupScript(src, timeout, functor) {
-  return new Promise((resolve, reject) => {
+export function setupScript(src, timeout) {
+  return setupScriptInternal(src, timeout);
+}
+
+function setupScriptInternal(src, timeout, attemptCount, existingTimer) {
+  attemptCount = attemptCount || 1;
+  if(attemptCount > 1) {
+    console.log("Re-attempting load of script '" + src + "': retry #" + attemptCount);
+  }
+
+  let promise = new Promise((resolve, reject) => {
     const Lscript = document.createElement('script');
     let hasResolved = false;
     Lscript.charset="UTF-8";        // KMEW-89
     Lscript.type = 'text/javascript';
     Lscript.async = false;
 
-    const timer = window.setTimeout(() => {
-      reject("Script load attempt timed out.");
+    const timer = existingTimer !== undefined ? existingTimer : window.setTimeout(() => {
+      // May have already been removed if there was an error -> repeat attempt.
+      Lscript.parentElement?.removeChild(Lscript);
+      hasResolved = true;
+      reject(new Error("Script load attempt for '" + src + "' timed out."));
     }, timeout);
 
     Lscript.onload = Lscript.onreadystatechange = () => {
+      Lscript.parentElement.removeChild(Lscript);
       window.clearTimeout(timer);
       if(!hasResolved && (Lscript.readyState === undefined || Lscript.readyState == "complete")) {
         hasResolved = true;
+
+        if(attemptCount > 1) {
+          console.log("Successfully loaded '" + src + "' after " + attemptCount + " attempts");
+        }
         resolve();
       }
     }
 
     Lscript.onerror = (err) => {
-      window.clearTimeout(timer);
-      reject(err);
+      // Cleanup the noisy script tag
+      Lscript.parentElement.removeChild(Lscript);
+
+      if(hasResolved) {
+        return;
+      }
+
+      // WARNING: err here is a very basic event object that doesn't actually give us useful info.
+
+      // One common side-effect of BrowserStack instability is that this event will be raised.
+      // Sadly, it's hard to know _why_ it's raised due to the point above - no useful info shows
+      // up in the logs, and often only one attempt fails.  But that may be key - _often only ONE
+      // attempt_ fails.  Therefore... let's enact a multi-retry on the script-load, so long as
+      // we don't exceed the timeout as a result!
+      if(attemptCount <= 3) {
+        window.setTimeout(() => {
+          let retryPromise = setupScript(src, timeout, attemptCount + 1, existingTimer);
```

Comment:
```suggestion
          let retryPromise = setupScriptInternal(src, timeout, attemptCount + 1, existingTimer);
```

With this change, `attemptCount` can get higher than 1...

Reply:
Oh, right.  Splitting it off into a helper was one of the last things I did; I _would_ manage to miss that detail.